in-progress/8025-publishing-tx-objects/design.md

```diff
@@ -382,6 +398,49 @@ In the event of a "failed" transaction, the transaction will appear in the block
 
 Further, the transaction's fee will be set to zero.
 
+### Proving phases
+
+<!-- Editors: you can copy/paste the png from the repository into excalidraw to make edits. -->
+
+![Proving Phases](./proving-phases.png)
+
+#### Monopoly claim phase
+
+The beginning of each epoch is the "monopoly claim phase".
+
+This phase has a duration of $MC$ slots (e.g. 2 slots).
+
+The committee member who could submit the first proposal in epoch `n` can submit a prover commitment bond.
+
+This can be a transaction separate from the submission of their proposed block.
+
+Doing so grants monopoly rights to the rewards for submitting the proof of epoch `n-1`.
```

Comment:
TLDR: I think this is worth it, mainly because validators can wait a few slots to "test" if they've gotten lumped in with a bad committee before submitting a bond, but also it has improvements in the reorg cases!

Starting with the status quo, I think that prover marketplaces will optimistically start proving epoch `n`, so when `n+1` comes around the work will be largely complete. The downside in this situation is that multiple marketplaces could perform redundant work, but
1. The total amount of redundant work won't be that great assuming there are only a handful of competitive proving marketplaces
2. In the future, they could organize an out-of-protocol auction where the proposer in `n+1` can commit to a prover during epoch `n`. The bond there could then be much smaller than the true "prover commitment bond" (e.g. 1/2 the actual fee they're going to pay to the prover marketplace)

I'm also a bit reluctant to have proposer post the bond for an unseen epoch because of the case where they are an honest node in a dishonest committee; we'd be forcing, for example, someone to make a blind bet of $1M that they don't roll a 1/million and get stuck with someone else willing to lose $10M.

But maybe that's an acceptable risk, especially since we have $C$ slots for the bond to land, so cautious validators might only submit prover commitment bonds if they get the option to after they've seen e.g. 3 valid proposals in the epoch.

The other benefit is that we tighten the max re-org window to just $E + P$, since the proof claim period for epoch `n` would coincide with the proof production period for `n-1`.

Further, in the case where no one submits the proof commitment bond, we can just re-org the epoch early and draw a new committee, which is a definite improvement as well.

Reply:
>  I think that prover marketplaces will optimistically start proving epoch n, so when n+1 comes around the work will be largely complete

I'm not sure about this. Unless we incentivize provers to submit proofs _fast_, there's no incentive to do that and waste computing power they could've directed elsewhere. We can validate it with the provers in the telegram group though!

>The other benefit is that we tighten the max re-org window 

Agree, this is another nice side-effect, aside from having much lower latency for L2 to L1 messages in the happy path.

> I'm also a bit reluctant to have proposer post the bond for an unseen epoch because of the case where they are an honest node in a dishonest committee; we'd be forcing, for example, someone to make a blind bet of $1M that they don't roll a 1/million and get stuck with someone else willing to lose $10M.

That's a good point. What do you think about still allowing the prover to submit the commitment bond at the end of the epoch? So the flow would be something like:

- Block 0 (or -1?): Proposer posts first block in the epoch, including a statement that "prover P has exclusive rights to this epoch". Prover P then starts proving. 
- Block 32: Prover P (**not** a proposer), after seeing all blocks in the epoch, posts the commitment bond signalling that the proof is coming up soon.

In the unhappy path, if no commitment bond is posted on block 32, we allow any prover to post a commitment bond for a few blocks before reorg'ing.

Reply:
Yes, I think it's pretty similar to the current design, only that a given prover has exclusive rights to posting the bond for a short period of time.